packages/x-date-pickers-pro/src/internal/hooks/useDesktopRangePicker/useDesktopRangePicker.tsx

```diff
@@ -173,6 +174,15 @@ export const useDesktopRangePicker = <
     },
   };
 
+  const componentsPropsForLayout: PickersViewLayoutSlotsComponentsProps = {
+    ...componentsProps,
+    toolbar: {
+      ...componentsProps?.toolbar,
+      currentlySelectingRangeEnd: currentDatePosition,
```

Comment:
Should we update the name of those props now or later ?
They are based on the v5 naming (`currentlySelectingRangeEnd` should become `currentDatePosition`, same for the setter)

Reply:
My main complain with the `currentlySelectingRangeEnd` naming is that I would expect a boolean, not a position.
But my new name is maybe not great.
It is never used on public API so if we want to change it, now is the time.
`currentDateRangePosition` or `currentPositionInDateRange` would probably be more explicit.

Reply:
Or you turn it into an actual boolean and name it `isEditingEndOfRange`

Reply:
Hmm... Boolean is always less flexible.
What if we will want/need additional positions along the line?
Like `startDate`, `startTime`, `endDate` and `endTime` 🤔 
Just extrapolating, that would probably not be the case, but if "enum" is working for now, I'd stick with it.

Reply:
True

Reply:
I am proposing simplfy `rangePosition` and to only have `start` and `end` in it (even for time ranges and date time ranges).
And to use the view to distinguish the start date and start time for instance.

I'm not using `dateRangePosition` because it is available on internals that will be used by `TimeRangePicker`, which is a little weird.

Reply:
Yeah, `rangePosition` should work as well. 👍 
The only small issue is that this component has it's props exposed via `themeAugmentation -> defaultProps` although, it might not be an issue for anyone to warrant a codemod, because they are required props and are always defined in our components.
The only issue could be if someone used the toolbar component for something externally. 🤔 
However small this chance could be, maybe it's worth noting this change in migration doc?

Reply:
I is also breaking it they provide a custom toolbar, which is a valid use case.
I will list it in the breaking changes :+1: